### PR TITLE
Add templating to porter config file

### LINKED ISSUE #1763: Allow using secrets and env vars in Porter's config file
When I specify the connection string or any sensitive values in Porter's config file, I want to be able to reference a secret or environment variable instead of hard-coding them in the config file. The goal is to eliminate sensitive values from the config file.

The azure plugin already reads from env vars, but it seems like a burden to make each plugin solve this problem, or does it help solve resolving secrets.

Here are a couple ideas for referencing an env var or secret:

**Smoosh in a protocol**
pros: you edit the same url entry as you did before
cons: it could conflict with the format required by the plugin. not easy to autocomplete
```toml
[[storage]]
  name = "cosmos"
  plugin = "mongodb"

  [storage.config]
    url = "secret://my-secret-with-a-connstr"
    url = "env://MYENV
```

**Add extra config prefixes**
pros: 🤷‍♀️
cons: could conflict with plugin's config keys they want to use
```toml
[[storage]]
  name = "cosmos"
  plugin = "mongodb"

  [storage.config]
    secret-url = "my-secret-with-a-connstr"
    env-url = "MYENV"
```

**Specify replacements**
pros: very specific, no chance of conflicting
cons: extra config sections, may be more confusing to people
```toml
[[storage]]
  name = "cosmos"
  plugin = "mongodb"

  [storage.secrets]
    url = "my-secret-with-a-connstr"

  [storage.env]
    url = "MYENV"
```

**Specify replacements**
pros: very specific, no chance of conflicting
cons: extra config sections, may be more confusing to people
```toml
[[storage]]
  name = "cosmos"
  plugin = "mongodb"

  [[storage.replacements]]
    type = "secret"
    url = "my-secret-with-a-connstr"

  [[storage.replacements]]
    type = "env"
    url = "MYENV"
```

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
Support templating in the Porter config file, e.g.
~/.porter/config.json|toml|yaml.

The template syntax uses a yaml-friendly delimiiter, ${}. For example,
${env.NAME} or ${secret.NAME}. If you are using toml or json for your
config file format, then you may need to use quotes around template
values to ensure that the config file has valid syntax when it is
loaded.

We load the config file first with only environment variables
substituted, then we initialize the secrets plugin, and finally do a
second pass of the config file to replace any secrets. So that first
pass needs to be a valid file based on the current format.

For example, in a porter.toml file, secrets should be quoted:

```toml
[[storage]]
  name = "dev"
  plugin = "mongodb"

  [storage.config]
    url = "${secret.connection-string}"
```

Only environment variables and secrets can be substituted. The secrets
are resolved from the default secret storage.

I am implementing this with the liquid template engine (from shopify
that was ported to Go). I don't want us to commit to supporting liquid
templates at this time, it's just an implementation detail that liquid
is used at the moment and it could change later.

Preview of the config documentation at https://deploy-preview-1879--porter.netlify.app/configuration/

# What issue does it fix
Closes #1763 

# Notes for the reviewer
Eventually porter.yaml will transition to this new template delimiter before v1.

# Checklist
- [x] Did you write tests?
- [ ] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://porter.sh/src/CONTRIBUTORS.md

### CODE REVIEW

File: docs/content/configuration.md [hunk 1]
```diff
@@ -3,99 +3,171 @@ title: Configuration
...
+# When default-secrets is not set, use the kubernetes.secret plugin.
+# This mode does not support additional configuration for the plugin.
+# If the plugin requires configuration, use default-secrets and define
+# the configuration in the secrets section.
+default-secrets-plugin = "kubernetes.secret"
+
+# Defines storage accounts
+[[storage]]
+  # The storage account name
+  name = "cosmos"
```
Comment (vdice) [inline docs/content/configuration.md:103]: Should/can this be `"devdb"` to match the `default-storage` value above?

File: docs/content/configuration.md [hunk 2]
```diff
@@ -212,4 +284,62 @@ experimental = ["structured-logs"]
...
+```toml
+output = "json"
+```
+
+### Allow Docker Host Access
+
+\--allow-docker-host-access controls whether the local Docker daemon or host should be made available to executing bundles.
+It is set with the PORTER_ALLOW_DOCKER_HOST_ACCESS environment variable.
+
+This flag is available for the following commands: install, upgrade], invoke, and uninstall.
```
Comment (vdice) [inline docs/content/configuration.md:337]:
```suggestion
This flag is available for the following commands: install, upgrade, invoke, and uninstall.
```

**Review by vdice — APPROVED** [review]
LGTM!  Excited to use this new functionality 👏

File: docs/content/configuration.md (hunk 1 above)
Comment (carolynvs) [reply docs/content/configuration.md:103]: oops!

**Pull request merged**